src/main/drivers/tim_def.h

```diff
@@ -0,0 +1,357 @@
+/*
+ * This file is part of Cleanflight.
+ *
+ * Cleanflight is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Cleanflight is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Cleanflight.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include <platform.h>
+#include "common/utils.h"
+
+#if defined(STM32F3)
+
+#define DEF_TIM(tim, chan, pin, flags, out) {\
```

Comment:
Whis will work fine now (DMA and AF is hidden in preprocessor tables), but in long-term these values can be inferred from simpler tables 
- DMA driver may provide channel -> handler mapping (possibly using something like IO _tag to refer to DMA channel and using 'object' functions to get required pieces of information)
- IO -> AF list may be described for each timer channel (ADC driver will need similar mapping
- N-channel may be inferred too (from tim+chan+io)

But this can be done in future, without affecting code outside timer implementation